src/components/AbuseReportForm/AbuseReportForm.tsx

```diff
@@ -97,97 +97,144 @@ export const Form: React.FC<{
   errors,
   pillar
 }) => {
-  const labelStylesClass = labelStyles(pillar);
-  return (
-    <form className={formWrapper}>
-      <div
-        className={css`
-          position: absolute;
-          right: ${space[3]}px;
-          top: ${space[3]}px;
-        `}
-      >
-        <Button
-          size="small"
-          iconSide="right"
-          icon={<SvgClose />}
-          onClick={toggleSetShowForm}
-        />
-      </div>
+  useEffect(() => {
+    const keyListener = (e: KeyboardEvent) => {
+      const listener = keyListenersMap.get(e.keyCode);
+      return listener && listener(e);
+    };
 
-      {errors && errors.response && (
-        <span
-          className={css`
-            color: red;
-          `}
-        >
-          {errors.response}
-        </span>
-      )}
+    document.addEventListener("keydown", keyListener);
 
-      <div className={inputWrapper}>
-        <label className={labelStylesClass} htmlFor="category">
-          Category
-        </label>
-        <select
-          name="category"
-          id="category"
-          onChange={categoryOnChange}
-          value={selectedCategory}
-        >
-          <option selected value="0">
-            Please select
-          </option>
-          <option value="1">Personal abuse</option>
-          <option value="2">Off topic</option>
-          <option value="3">Legal issue</option>
-          <option value="4">Trolling</option>
-          <option value="5">Hate speech</option>
-          <option value="6">Offensive/Threatening language</option>
-          <option value="7">Copyright</option>
-          <option value="8">Spam</option>
-          <option value="9">Other</option>
-        </select>
-        {errors && errors.category && (
+    return () => document.removeEventListener("keydown", keyListener);
+  });
+
+  const modalRef = createRef<HTMLDivElement>();
+  useEffect(() => {
+    if (!modalRef.current) return;
+    const closeModal = modalRef.current.querySelector(
+      'select[name="category"]'
+    );
+    closeModal && closeModal.focus();
+  }, [modalRef]);
+
+  const handleTabKey = (e: KeyboardEvent) => {
+    if (!modalRef.current) return;
+    const focusableModalElements = modalRef.current.querySelectorAll(
+      'select[name="category"], textarea[name="reason"], input[type="email"], button[type="submit"], button[custom-guardian="close-modal"]'
+    );
+    const firstElement = focusableModalElements[0];
+    const lastElement =
+      focusableModalElements[focusableModalElements.length - 1];
+
+    if (!e.shiftKey && document.activeElement === lastElement) {
+      firstElement.focus();
+      return e.preventDefault();
+    }
+
+    if (e.shiftKey && document.activeElement === firstElement) {
+      lastElement.focus();
+      e.preventDefault();
+    }
+  };
```

Comment:
This is pretty narly. We could help future developers with comments explaining what the reasons and intentions are here with maybe some links to the supporting articles?